crates/ruff_linter/src/rules/ruff/rules/unit_test_super.rs

```diff
@@ -0,0 +1,104 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::identifier::Identifier;
+use ruff_python_ast::{self as ast, Arguments, Expr, Stmt, StmtClassDef};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for subclasses of unittest. `TestCase` where the `setUp()`, `tearDown()`,
+/// `setUpClass()`, or `tearDownClass()` methods do not call the corresponding `super()`
+/// method.
+///
+/// ## Why is this bad?
+/// Failing to call the corresponding `super()` method can lead to incomplete or
+/// improper initialization and cleanup.
+///
+///
+/// ## Example
+///
+/// ```python
+/// import unittest
+///
+///
+/// class MyTestCase(unittest.TestCase):
+///     def setUp(self):
+///         self.resource = "Test resource"
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// import unittest
+///
+///
+/// class MyTestCase(unittest.TestCase):
+///     def setUp(self):
+///         super().setUp()
+///         self.resource = "Test resource"
+/// ```
+///
+#[derive(ViolationMetadata)]
+pub(crate) struct UnitTestSuper;
+
+impl Violation for UnitTestSuper {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Call super methods for unit test methods".to_string()
+    }
+}
+
+pub(crate) fn unit_test_super(checker: &mut Checker, class: &StmtClassDef) {
+    let Some(Arguments { args: bases, .. }) = class.arguments.as_deref() else {
+        return;
+    };
+
+    if !is_testcase_subclass(bases) {
+        return;
+    }
+
+    let unit_test_methods = ["setUp", "tearDown", "setUpClass", "tearDownClass"];
+
+    for stmt in &class.body {
+        if let Stmt::FunctionDef(function_def) = stmt {
+            let body = &function_def.body;
+            let name = function_def.name.as_str();
+            if unit_test_methods.contains(&name) && !has_super_call(body, name) {
```

Comment:
Nit: I'd use a `match` statement here
```suggestion
            if matches!(name, "setUp" | "tearDown" | "setUpClass" | "tearDownClass) && !has_super_call(body, name) {
```